src/featured_graph.jl

```diff
@@ -0,0 +1,278 @@
+#===================================
+Define FeaturedGraph type as a subtype of LightGraphs' AbstractGraph.
+For the core methods to be implemented by any AbstractGraph, see
+https://juliagraphs.org/LightGraphs.jl/latest/types/#AbstractGraph-Type
+https://juliagraphs.org/LightGraphs.jl/latest/developing/#Developing-Alternate-Graph-Types
+=============================================#
+
+abstract type AbstractFeaturedGraph <: AbstractGraph{Int} end
+
+"""
+    NullGraph()
+
+Null object for `FeaturedGraph`.
+"""
+struct NullGraph <: AbstractFeaturedGraph end
+
+
+struct FeaturedGraph <: AbstractFeaturedGraph
+    edge_index
+    num_nodes::Int
+    num_edges::Int
+    # ndata::Dict{String, Any} # https://github.com/FluxML/Zygote.jl/issues/717        
```

Comment:
Is it possible to include various kind of features in a `FeaturedGraph`? I had think of this before, but I give up and turn to `MetaGraph`. `MetaGraph` has implemented similar structure as this. And another problem will come to we should provide a way for user to specify what node/edge/global feature them want to train on.

Is it better to use `Symbol` as key?

Reply:
This requires some extra thought, I don't plan to implement it in this PR, just wanted to write down a stub. 

Both PytorchGeometric and DGL support adding an arbitrary number of edge and node features, we should do the same at some point since it is something very convenient to have (e.g. you want to store node labels, training masks etc...)

MetaGraph.jl is not a good fit. Besides being scarcely maintained, it provides maps with single nodes/edges as keys which is not good for vectorized operations, we want to store arrays.